apps/web/src/hooks/use-pagination.js

```diff
@@ -0,0 +1,52 @@
+import { useState } from 'react'
+
+const SORTS = ['asc', 'desc', undefined]
+
+const usePagination = (
+  initialPage = 1,
+  initialSortBy,
+  initialSortDirection = 'asc'
+) => {
+  const [pagination, _setPagination] = useState({
+    page: initialPage,
+    sortBy: initialSortBy,
+    sortDirection: initialSortDirection,
+  })
+
+  const setPagination = (newPagination) => {
+    _setPagination({ ...pagination, ...newPagination })
+  }
+
+  const setPage = (page) => setPagination({ page })
+  const setSortBy = (sortBy) => setPagination({ sortBy })
+  const setSortDirection = (sortDirection) => setPagination({ sortDirection })
+
+  const handleSortBy = (newSortBy) => {
+    const nextSortDirection =
+      pagination.sortBy === newSortBy
+        ? SORTS[(SORTS.indexOf(pagination.sortDirection) + 1) % 3]
```

Comment:
Trying to understanding what's happening here - is dividing by 3 required due to the sort options? Should this be divided by `SORTS` length?

Reply:
I was able to sort successfully, so I know it works, just curious :)